app/javascript/controllers/area_served_controller.js

```diff
@@ -0,0 +1,28 @@
+import { Controller } from "@hotwired/stimulus"
+export default class extends Controller {
+    static targets= ["share", "total", "warning"]
+    connect () {
+        this.calculateClientShareTotal()
+    }
+    calculateClientShareTotal(){
+        let total = 0;
+        let share_targets = this.shareTargets
+        share_targets.forEach( share_target =>{
+                if(share_target.value){
+                    total += parseInt(share_target.value);
+                }
+        }
+        )
+
+        this.totalTarget.innerHTML = total + " %"
+
+        if(total == 0 || total == 100){
```

Comment:
Also error if any one of them is < 0 or > 100

Reply:
Or I could just make it a selection from 1-100.

Reply:
From a personal taste point of view, I prefer the appearance of numeric.   Just preventing them from entering wrong data at all seems to be better, though.     Unfortunately, it doesn't look like simpleform knows enough to handle "it has to be whole number numeric *and* in a range"